src/sentry/runner/initializer.py

```diff
@@ -325,6 +325,8 @@ def initialize_app(config: dict[str, Any], skip_service_validation: bool = False
 
     bootstrap_options(settings, config["options"])
 
+    logging.raiseExceptions = not settings.IS_PROD
```

Comment:
does this do anything?

Reply:
AFAIK it does affect the built-in logging backends, it just didn't do anything for our custom one without the changes. I'm gonna add some tests to validate.

Reply:
verified: https://github.com/getsentry/sentry/pull/60131/files#diff-e44d3e9afba2ea3b5ea017ad24462b4760740fdbe7c7f7f03067426e67be6fbeR100-R114